app/scripts/metamask-controller.js

```diff
@@ -7205,14 +7222,14 @@ export default class MetamaskController extends EventEmitter {
 
     this.tokenListController.updatePreventPollingOnNetworkRestart(!newEnabled);
```

Comment:
I've never understood the purpose of `preventPollingOnNetworkRestart`, so I don't know whether it still makes sense.  When true, all it seems to do is reset the state when switching chains.